lib/index.js

```diff
@@ -11,6 +11,16 @@ exports.Service = class Service {
 
         this.server = server;
         this.options = options;
+        const methods = new Set(Object.getOwnPropertyNames(Object.getPrototypeOf(this))
```

Comment:
Ideally all methods up the prototype chain would be bound, not just methods on the immediate class.

Reply:
@devinivy Actually my first pass at this did just that, but I thought I was being overly cautious and thought that a user would only ever use the methods defined in the immediate class. But this affirms my initial feelings. Def agree with this 👍